python/templates/Interface.h.jinja2

```diff
@@ -89,10 +89,9 @@ class {{ class.bare_type }} {
  std::unique_ptr<Concept> m_self{nullptr};
 
 public:
-  template<typename ValueT>
+  template<typename ValueT, typename = std::enable_if_t<isInitializableFrom<ValueT>>>
   {{ class.bare_type }}(ValueT value) :
     m_self(std::make_unique<Model<podio::detail::GetDefaultHandleType<ValueT>>>(value)) {
-    static_assert(isInitializableFrom<ValueT>, "{{ class.bare_type }} can only be initialized with one of the following types (and their Mutable counter parts): {{ Types | join(", ") }}");
```

Comment:
Wondering whether we should leave this in, but I don't think it's visible in the compiler error message in any case, right?

Reply:
I'm not sure if there is any scenario in which that would get printed- even if some would try to explicitly instantiate wrong overload the compiler will print some substitution error in the `enable_if`.

Reply:
Yeah, they are rather terse with context around where things happen. It would only help if people look into the header file then and see the message to the `static_assert`. In principle `isInitializableFrom<ValueT>` is rather obvious I think, but you have to find it (and be able to interpret it), in all the compiler output.

Reply:
Hmm, the comment could be moved to line with enable_if so it appears somewhere in the compilation error
```
/cvmfs/sw.hsf.org/contrib/x86_64-almalinux9-gcc11.4.1-opt/gcc/14.2.0-yuyjov/lib/gcc/x86_64-pc-linux-gnu/14.2.0/../../../../include/c++/14.2.0/type_traits: In substitution of 'template<bool _Cond, class _Tp> using std::enable_if_t = typename std::enable_if::type [with bool _Cond = false; _Tp = void]':
/home/mafila/podio/tests/datamodel/TypeWithEnergy.h:96:29:   required from here
   96 |   template<typename ValueT, typename = std::enable_if_t<isInitializableFrom<ValueT>>> // TypeWithEnergy can only be initialized with one of the following types (and their Mutable counter parts): ExampleHit, ExampleMC, ExampleCluster");
      |                             ^~~~~~~~
/cvmfs/sw.hsf.org/contrib/x86_64-almalinux9-gcc11.4.1-opt/gcc/14.2.0-yuyjov/lib/gcc/x86_64-pc-linux-gnu/14.2.0/../../../../include/c++/14.2.0/type_traits:2696:11: error: no type named 'type' in 'struct std::enable_if<false, void>'
 2696 |     using enable_if_t = typename enable_if<_Cond, _Tp>::type;
```
Would it be helpful?

Reply:
It's definitely better than nothing, and it will appear pretty much at the top somewhere, right?

Reply:
I think, it's pretty much the best we can do at this point without concepts.